src/formatDate.js

```diff
@@ -9,6 +9,57 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const divider = fromFormat[3];
+  const dividerNew = toFormat[3];
+  const parts = date.split(divider);
+
+  const fromFormatMap = {};
+
+  fromFormat.forEach((format, index) => {
+    fromFormatMap[format] = parts[index];
+  });
+
+  const { YYYY, YY, MM, DD } = fromFormatMap;
```

Comment:
Destructuring directly from 'fromFormatMap' is a neat way to extract the date parts. However, ensure that the keys used for destructuring match the expected format elements exactly, including cases where they might not be present.